### ISSUE TITLE
Use consistent line endings. Remove trailing whitespaces. Use .editorconfig.

### ISSUE THREAD

**Issue by MrSmith33** [gecko0307/dlib#129]
It is quite tedious to fix two lines of code and see 200 lines diff, because of different line endings or extra spaces that wheren't deleted by mistake.

Personally I use following `.editorconfig` file. Check [details here](http://editorconfig.org/)
```
root = true

[*.{c,h,d,di,dd,json}]
end_of_line = lf
insert_final_newline = true
indent_style = tab
indent_size = 4
trim_trailing_whitespace = true
charset = utf-8
```

If you approve I can fix all the files at once.

**Comment by gecko0307** [comment]
I've fixed trailing spaces in 6ad98cf.
I prefer CRLF and 4-space indents. Does it cause problems?

**Comment by MrSmith33** [comment]
Any option is good if it is used consistently.

**Comment by MrSmith33** [comment]
Applying that editorconfig gives following diff. (plus manually fixed one tab)
[diff.zip](https://github.com/gecko0307/dlib/files/706552/diff.zip)

**Comment by gecko0307** [comment]
There are lots of LF -> CRLF conversions, so the diff obviously should be huge.

**Comment by MrSmith33** [comment]
Should I send pull request?

**Comment by gecko0307** [comment]
It's strange that LFs are still appearing, though - I've already converted them. Are you using latest repository state? I've noticed that in some cases after a long period of pulling a local copy ends up inconsistent with remote repo, while git says 'Already up-to-date'. Maybe do a fresh `git clone`?

**Comment by MrSmith33** [comment]
I havent committed those changes yet

**Comment by MrSmith33** [comment]
Closed by f1dc3bed2710b1c6a55ad5117b8060c85b21c378
